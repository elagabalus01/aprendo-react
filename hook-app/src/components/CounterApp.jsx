import {useState} from "react";

export const CounterApp = ()=>{

    const [counter, setCounter] = useState({
        counter1: 1,
        counter2: 2,
        counter3: 3
    })


    return (
        <>
            <h1>Counter app</h1>
            <h2>{counter.counter1}</h2>
            <h2>{counter.counter2}</h2>
            <h2>{counter.counter3}</h2>

            <button className={"btn"} onClick={()=>{
                setCounter({
                    counter1: counter.counter1 +1,
                    counter2: counter.counter2,
                    counter3: counter.counter3
                })
            }}> +1 </button>

        </>
    )


}
