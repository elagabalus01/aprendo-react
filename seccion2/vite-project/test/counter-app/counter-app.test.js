import {fireEvent, render, screen} from "@testing-library/react";
import CounterApp from "../../src/counter-app/counter-app.jsx";

describe("PRUEBAS SOBRE COUNTER APP", ()=>{

    const valorInicial = 100

    test("debe hacer match con el snapshot", ()=>{
        const {container}= render(<CounterApp></CounterApp>)
        expect(container).toMatchSnapshot()
    })

    test("debe mostrar e valor inicial de 100", ()=>{
        const {getAllByText}= render(<CounterApp value={valorInicial}></CounterApp>)
        expect(screen.getByText(`valor actual ${valorInicial}`)).toBeTruthy()
    })

    test("debe incrementar con el boton +1", ()=>{
        render(<CounterApp value={valorInicial}></CounterApp>)
        fireEvent.click(screen.getByText('1'))

        expect(screen.getByText(`valor actual ${valorInicial + 1}`)).toBeTruthy()

    })


    test("debe decrementar con el boton -1", ()=>{
        render(<CounterApp value={valorInicial}></CounterApp>)
        fireEvent.click(screen.getByText('toggle'))
        screen.debug()
        fireEvent.click(screen.getByText('1'))
        expect(screen.getByText(`valor actual ${valorInicial - 1}`)).toBeTruthy()

    })

    test("Reset", ()=>{
        render(<CounterApp value={valorInicial}></CounterApp>)
        fireEvent.click(screen.getByText('1'))
        fireEvent.click(screen.getByText('1'))
        fireEvent.click(screen.getByText('1'))
        fireEvent.click(screen.getByText('1'))
        //fireEvent.click(screen.getByText('reset'))

        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}))

        expect(screen.getByText(`valor actual ${valorInicial}`)).toBeTruthy()

    })

})
