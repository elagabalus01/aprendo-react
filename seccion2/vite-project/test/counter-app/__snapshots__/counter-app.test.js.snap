// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PRUEBAS SOBRE COUNTER APP debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Countador
  </h1>
  <h1>
    valor actual 
  </h1>
  <button>
    1
  </button>
  <button>
    toggle
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
</div>
`;
